include/vcpkg/postbuildlint.buildtype.h

```diff
@@ -5,61 +5,32 @@
 #include <vcpkg/build.h>
 
 #include <array>
-#include <regex>
 
 namespace vcpkg::PostBuildLint
 {
     struct BuildType
     {
-        enum class BackingEnum
-        {
-            DEBUG_STATIC = 1,
-            DEBUG_DYNAMIC,
-            RELEASE_STATIC,
-            RELEASE_DYNAMIC
-        };
-
-        static BuildType value_of(const Build::ConfigurationType& config, const Build::LinkageType& linkage);
-
         BuildType() = delete;
 
-        constexpr BuildType(const BackingEnum backing_enum,
-                            const Build::ConfigurationType config,
-                            const Build::LinkageType linkage)
-            : backing_enum(backing_enum), m_config(config), m_linkage(linkage)
+        constexpr BuildType(Build::ConfigurationType c, Build::LinkageType l)
+            : config(c), linkage(l)
         {
         }
 
-        constexpr operator BackingEnum() const { return backing_enum; }
+        bool has_crt_linker_option(StringView sv) const;
+        StringLiteral to_string() const;
 
-        const Build::ConfigurationType& config() const;
-        const Build::LinkageType& linkage() const;
-        const std::regex& crt_regex() const;
-        const std::string& to_string() const;
+        static BuildType VALUES[4];
```

Comment:
I think it would be better to have this static in the .cpp.

Reply:
I don't understand this - what do you mean? It is in the .cpp.